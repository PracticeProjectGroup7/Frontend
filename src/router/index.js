import { createRouter, createWebHistory } from 'vue-router';

import HomeView from '../views/HomeView.vue';
import LoginView from '../views/accounts/LoginView.vue';
import RegisterView from '../views/accounts/RegisterView.vue';
import BookingHistoryView from '../views/BookingHistory.vue';
import BookingBillView from '../views/BookingBill.vue';

const ROUTE_HOME = 'ROUTE_HOME';
const ROUTE_LOGIN = 'ROUTE_LOGIN';
const ROUTE_REGISTER = 'ROUTE_REGISTER';
const ROUTE_LOGIN_INTERNAL = 'ROUTE_LOGIN_INTERNAL';
const ROUTE_BOOKING_HISTORY_SELF = 'ROUTE_BOOKING_HISTORY_SELF';
const ROUTE_BOOKING_HISTORY_OTHERS = 'ROUTE_BOOKING_HISTORY_OTHERS';
const ROUTE_BOOKING_BILL = 'ROUTE_BOOKING_BILL';

const routes = [
  {
    path: '/',
    name: ROUTE_HOME,
    component: HomeView,
  },
  {
    path: '/login',
    name: ROUTE_LOGIN,
    component: LoginView,
  },
  {
    path: '/register',
    name: ROUTE_REGISTER,
    component: RegisterView,
  },
  {
    path: '/_internal/login',
    name: ROUTE_LOGIN_INTERNAL,
    component: LoginView,
    props: { internal: true },
  },
  {
    path: '/booking-history',
    name: ROUTE_BOOKING_HISTORY_SELF,
    component: BookingHistoryView,
  },
  {
    path: '/booking-history/:patientId',
    name: ROUTE_BOOKING_HISTORY_OTHERS,
    component: BookingHistoryView,
    props: true,
  },
  {
    path: '/booking-bill/:billId',
    name: ROUTE_BOOKING_BILL,
    component: BookingBillView,
    props: true,
  },
];

const router = createRouter({
  history: createWebHistory(import.meta.env.BASE_URL),
  routes: routes,
});

export {
  routes,
  ROUTE_HOME,
  ROUTE_LOGIN, ROUTE_LOGIN_INTERNAL, ROUTE_REGISTER,
  ROUTE_BOOKING_HISTORY_SELF, ROUTE_BOOKING_HISTORY_OTHERS,
  ROUTE_BOOKING_BILL,
};

export default router;
