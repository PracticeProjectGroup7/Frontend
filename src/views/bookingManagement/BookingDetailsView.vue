<script setup>
const FILENAME = 'BookingDetailsView.vue';
import { ref, computed } from 'vue';
import { RouterLink } from 'vue-router';
import { isDoctorType } from '../../utils/utils';
import { doctorBooking, labBooking } from '../../_dummy_data/bookings';
import { ROUTE_BOOKING_HISTORY_OTHERS } from '../../router';
import UpdateBookingStatusModal from '../../components/Modals/UpdateBookingStatusModal.vue';

const props = defineProps({
  'testId': String,
  'bookingType': String,
});
const showModal = ref(false);
const openModal = () => {
  showModal.value = true;
};
const isDoctorTypeBooking = isDoctorType(props.bookingType);

const bookingTypeStr = computed(() => {
  return isDoctorTypeBooking ? 'Appointment' : 'Test';
});
const breadcrumbPath = computed(() => {
  return isDoctorTypeBooking ? '/appointment-management' : '/test-management';
});

console.log(FILENAME, 'On details page');

const booking = ref(isDoctorTypeBooking ? doctorBooking : labBooking);

const handleStatusSaved = (data) => {
  // Handle the updated status data here
  console.log(FILENAME, 'Status saved:', data);
};

const closeModal = () => {
  showModal.value = false;
};

</script>

<template data-theme="corporate">
  <div class="p-8">
    <div class="text-sm breadcrumbs">
      <ul>
      <li><RouterLink :to="breadcrumbPath">
        {{ bookingTypeStr }} Management
      </RouterLink></li>
      <li>View {{ bookingTypeStr }} Details</li>
      </ul>
    </div>
    <div class="p-8">
      <div class="flex items-center"> <!-- Add a flex container -->
        <h2 class="text-xl font-semibold mb-2">
          {{ isDoctorTypeBooking ? 'Appointment' : 'Lab Test' }} Details - Booking #{{ booking.id }}
        </h2>
        <span
          :class="{
            'bg-orange-700': booking.status === 'pending',
            'bg-green-700': booking.status === 'completed',
          }"
          class="rounded-full py-1 px-2 text-white ml-4"
        >
          {{ booking.status }}
        </span>
      </div>
    </div>

    <div class="grid grid-cols-2 gap-2 ml-8">
      <div>
        <ul>
          <li><strong>Patient Name:</strong> {{ booking.patientName }}</li>
          <li><strong>Date of Birth:</strong> {{ booking.dob }}</li>
          <li><strong>Gender:</strong> {{ booking.gender }}</li>
        </ul>
      </div>

      <div>
        <ul>
          <li>
            <strong>
              {{ bookingTypeStr }} Name:
            </strong>
            {{ isDoctorTypeBooking ? booking.doctorName : booking.testName }}
          </li>
          <li>
            <strong>
              {{ bookingTypeStr }} Date:
            </strong>
            {{ isDoctorTypeBooking ? booking.appointmentDate : booking.testDate }}
          </li>
          <li>
            <strong>
              {{ bookingTypeStr }} Result:
            </strong>
            {{ isDoctorTypeBooking ? booking.diagnosis : booking.testResult }}
          </li>
        </ul>
      </div>
    </div>

    <div class="mt-4 space-x-4 p-8">
      <RouterLink :to="{ name: ROUTE_BOOKING_HISTORY_OTHERS, params: { patientId: booking.patientId } }">
        <button class="view-history-button">View Appointment History</button>
      </RouterLink>
      <button class="update-status-button" @click="openModal">Update Status</button>
    </div>
  </div>
  <UpdateBookingStatusModal
      v-if="showModal"
      :booking="booking"
      :isDoctorTypeBooking="isDoctorTypeBooking"
      @status-saved="handleStatusSaved"
      @close="closeModal"
    />
</template>

<style scoped>
li {
  @apply mb-2;
}

.view-history-button,
.update-status-button {
  @apply bg-white text-black border border-black px-4 py-2 rounded cursor-pointer transition-colors duration-300;
}

.view-history-button:hover,
.update-status-button:hover {
  @apply bg-black text-white;
}
</style>
