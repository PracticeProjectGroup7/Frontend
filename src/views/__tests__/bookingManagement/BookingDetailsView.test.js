import { vi, test, beforeEach, expect } from 'vitest';
import { createRouter, createWebHistory } from 'vue-router';
import { createTestingPinia } from '@pinia/testing';
import { routes } from '../../../router';
import { mount } from '@vue/test-utils';
import BookingDetailsView from '../../bookingManagement/BookingDetailsView.vue';

let router;

beforeEach(async () => {
  router = createRouter({
    history: createWebHistory(),
    routes: routes,
  });
});

test('Renders BookingDetailsView component', async () => {
  // Push a router state to simulate a route
  router.push('/test-management'); // Replace with the appropriate route
  await router.isReady();

  const wrapper = mount(BookingDetailsView, {
    global: {
      plugins: [
        createTestingPinia({
          createSpy: vi.fn, // Use createSpy from Vitest
        }),
      ],
    },
    mocks: {
      $route: {
        params: {
          bookingId: 'your-booking-id', // Replace with a valid booking ID
        },
      },
    },
  });

  await wrapper.vm.$nextTick(); // Wait for the next tick (update) to complete

  // Add specific assertions here
  // For example, check if important elements or data is rendered
  expect(wrapper.element).toMatchSnapshot(); // Use snapshot testing if available
});
