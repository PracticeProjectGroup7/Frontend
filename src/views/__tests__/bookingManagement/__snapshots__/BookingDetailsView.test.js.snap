// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders BookingDetailsView component 1`] = `
<div
  data-v-app=""
>
  
  <!--v-if-->
  <!--v-if-->
  
</div>
`;
